common/src/main/java/org/apache/sedona/common/geography/Constructors.java

```diff
@@ -264,4 +264,271 @@ private static Geometry collectionToGeom(Geography g, GeometryFactory gf) {
+      return lineToGeog(geom);
+    } else if ("MultiLineString".equalsIgnoreCase(type)) {
+      return mLineToGeog(geom);
+    } else if ("Polygon".equalsIgnoreCase(type)) {
+      return polyToGeog(geom);
+    } else if ("MultiPolygon".equalsIgnoreCase(type)) {
+      return mPolyToGeog(geom);
+    } else if ("GeometryCollection".equalsIgnoreCase(type)) {
+      return geomCollToGeog(geom);
+    } else {
+      throw new UnsupportedOperationException("Unsupported geometry type: " + type);
+    }
+  }
+
+  private static Geography pointToGeog(Geometry geom) throws IOException {
+    Coordinate[] pts = geom.getCoordinates();
+    if (pts.length == 0 || Double.isNaN(pts[0].x) || Double.isNaN(pts[0].y)) {
+      return new SinglePointGeography();
+    }
+    double lon = pts[0].x;
+    double lat = pts[0].y;
+
+    S2Point s2Point = S2LatLng.fromDegrees(lat, lon).toPoint();
+    S2Builder builder = new S2Builder.Builder().build();
+    S2PointVectorLayer layer = new S2PointVectorLayer();
+    builder.startLayer(layer);
+    builder.addPoint(s2Point);
+
+    S2Error error = new S2Error();
+    if (!builder.build(error)) {
+      throw new IOException("Failed to build S2 point layer: " + error.text());
+    }
+    List<S2Point> points = layer.getPointVector();
+    if (points.isEmpty()) {
+      return new SinglePointGeography();
+    }
+    return new SinglePointGeography(points.get(0));
+  }
+
+  private static Geography mPointToGeog(Geometry geom) throws IOException {
+    Coordinate[] pts = geom.getCoordinates();
+    if (pts.length == 0 || Double.isNaN(pts[0].x) || Double.isNaN(pts[0].y)) {
+      return new PointGeography();
+    }
+    List<S2Point> points = new ArrayList<>(pts.length);
+    // Build via S2Builder + S2PointVectorLayer
+    S2Builder builder = new S2Builder.Builder().build();
+    S2PointVectorLayer layer = new S2PointVectorLayer();
+    // must call build() before reading out the points
+    S2Error error = new S2Error();
+    for (int i = 0; i < pts.length; i++) {
+      double lon = pts[i].x;
+      double lat = pts[i].y;
+      S2Point s2Point = S2LatLng.fromDegrees(lat, lon).toPoint();
+
+      builder.startLayer(layer);
+      builder.addPoint(s2Point);
+
+      if (!builder.build(error)) {
+        throw new IOException("Failed to build S2 point layer: " + error.text());
+      }
+    }
+    for (int i = 0; i < layer.getPointVector().size(); i++) {
+      // Extract the resulting points
+      points.add(layer.getPointVector().get(i));
+    }
+    return new PointGeography(points);
+  }
+
+  private static Geography lineToGeog(Geometry geom) throws IOException {
+    LineString ls = (LineString) geom;
+
+    // Build S2 points
+    List<S2Point> pts = toS2Points(ls.getCoordinates());
+    if (pts.size() < 2) {
+      // empty or degenerate → empty single polyline
+      SinglePolylineGeography empty = new SinglePolylineGeography();
+      empty.setSRID(geom.getSRID());
```

Comment:
SRID is only handled in this place but not other places. We can set SRID in `geomToGeography` instead of setting it everywhere.